plugins/outputs/riemann_legacy/riemann.go

```diff
@@ -110,9 +110,9 @@ func buildEvents(p telegraf.Metric, s string) []*raidman.Event {
 			Service: serviceName(s, p.Name(), p.Tags(), fieldName),
 		}
 
-		switch value.(type) {
+		switch value := value.(type) {
 		case string:
-			event.State = value.(string)
+			event.State = value
```

Comment:
Please use an if statement instead
```golang
if v, ok := value.(string); ok {
    event.State = v
} else {
    event.Metric := value
}

Reply:
Hi @srebhan 

Can you please check my comment above? If you want, I can carry out a project-wide refactor for this instead of fixing a few instances here and there. Please let me know what you think about this.